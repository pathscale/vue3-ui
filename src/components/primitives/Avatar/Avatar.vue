<script>
/* eslint-disable no-undef -- Access to Benchie variable/function */
import { onBeforeMount, ref } from 'vue'

export default {
  name: 'VAvatar',
  props: {
    alt: String,
    size: {
      type: String,
      default: 'is-64x64',
    },
    rounded: {
      type: Boolean,
      default: true,
    },
    background: {
      type: String,
      default: 'has-background-link',
    },
    text: {
      type: String,
      default: 'has-text-white',
    },
    src: {
      type: String,
      default: null,
    },
    dataSrc: {
      type: String,
      default: null,
    },
    customClass: String,
  },
  setup(props) {
    const source = ref(props.src || props.dataSrc)
    const background = ref(props.background)
    const text = ref(props.text)
    const alt = ref(props.alt)

    onBeforeMount(async () => {
      if (props.dataSrc && hasBenchieSupport) {
        source.value = await t(props.dataSrc, $__CDN)
      }
    })

    const backgroundColor = source.value ? '' : background.value

    const textColor = source.value ? '' : text.value

    const caption = alt.value ? alt.value.split(' ')[0][0] + alt.value.split(' ')[1][0] : ""

    return { source, backgroundColor, textColor, caption }
  },
}
</script>

<template>
  <figure
    class="image is-flex is-justify-content-center is-align-items-center mx-1"
    :class="[size, backgroundColor, textColor]">
    <img
      :src="source"
      :data-src="dataSrc"
      :alt="alt"
      v-if="source"
      :class="[customClass, { 'is-rounded': rounded }]" />
    <span class="is-size-4" v-if="!source" :class="[{ 'is-rounded': rounded }]">
      {{ caption }}
    </span>
  </figure>
</template>