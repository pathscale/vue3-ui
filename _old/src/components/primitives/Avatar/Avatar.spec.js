import { shallowMount } from '@vue/test-utils'
import Avatar from './Avatar.vue'

let wrapper

describe('Avatar', () => {
  beforeEach(() => {
    wrapper = shallowMount(Avatar)
  })

  it('is called', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
