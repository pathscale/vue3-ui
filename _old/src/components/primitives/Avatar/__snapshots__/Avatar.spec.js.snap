// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar render correctly 1`] = `
<figure class="image is-flex is-justify-content-center is-align-items-center mx-1 is-64x64 has-background-link has-text-white">
  <!--v-if--><span class="is-size-4 is-rounded"></span>
</figure>
`;